mediators/datamapper/components/org.wso2.carbon.gateway.mediators.datamapper.engine/src/main/java/org/wso2/carbon/gateway/mediators/datamapper/engine/core/mapper/MappingResource.java

```diff
@@ -0,0 +1,146 @@
+     */
+    public MappingResource(InputStream inputSchema, InputStream outputSchema, InputStream mappingConfig,
+            String outputType) throws SchemaException, JSException {
+        this.inputSchema = getJSONSchema(inputSchema);
+        this.outputSchema = getJSONSchema(outputSchema);
+        this.inputRootelement = this.inputSchema.getName();
+        this.outputRootelement = this.outputSchema.getName();
+        this.propertiesList = new ArrayList<>();
+        this.function = createFunction(mappingConfig, outputType);
+    }
+
+    private Schema getJSONSchema(InputStream inputSchema) throws SchemaException {
+        return new JacksonJSONSchema(inputSchema);
+    }
+
+    public Schema getInputSchema() {
+        return inputSchema;
+    }
+
+    public Schema getOutputSchema() {
+        return outputSchema;
+    }
+
+    public JSFunction getFunction() {
+        return function;
+    }
+
+    public List getPropertiesList() {
+        return propertiesList;
+    }
+
+    /**
+     * Creates java script function by passing the configuration file.
+     * Since this function going to execute every time when message hit the mapping backend, 
+     * this function save in the resource model.
+     *
+     * @param mappingConfig mapping configuration
+     * @return java script function
+     */
+    private JSFunction createFunction(InputStream mappingConfig, String outputType) throws JSException {
+        BufferedReader configReader = new BufferedReader(new InputStreamReader(mappingConfig, StandardCharsets.UTF_8));
+        //need to identify the main method of the configuration because that method going to
+        // execute in engine
+        String[] inputRootElementArray = inputRootelement.split(NAMESPACE_DELIMETER);
+        String inputRootElement = inputRootElementArray[inputRootElementArray.length - 1];
+        String[] outputRootElementArray = outputRootelement.split(NAMESPACE_DELIMETER);
+        String outputRootElement = outputRootElementArray[outputRootElementArray.length - 1];
+        String jsFunctionBody;
+
+        String propertiesPattern = "(DM_PROPERTIES.)([a-zA-Z_$][a-zA-Z_$0-9]*)\\['([a-zA-Z_$][a-zA-Z-_.$0-9]*)'\\]";
+        Pattern pattern = Pattern.compile(propertiesPattern);
+        Matcher match;
+
+        String fnName =
+                FUNCTION_NAME_CONST_1 + inputRootElement + FUNCTION_NAME_CONST_2 + outputRootElement + BRACKET_OPEN
+                        + BRACKET_CLOSE;
+        if (InputOutputDataType.JSON.toString().equals(outputType)) {
+            fnName = JS_STRINGIFY + BRACKET_OPEN + fnName + BRACKET_CLOSE;
+        }
+        String configLine;
+        StringBuilder configScriptBuilder = new StringBuilder();
+        try {
+            while ((configLine = configReader.readLine()) != null) {
+                configScriptBuilder.append(configLine);
+            }
+        } catch (IOException e) {
+            throw new JSException(e.getMessage());
```

Comment:
i'm not sure about the context, shouldn't we be passing 'e' as well (i could be wrong) . what's the best practice?